plugins/resource_monitor_plugin/include/eosio/resource_monitor_plugin/file_space_handler.hpp

```diff
@@ -6,6 +6,8 @@
 #include <eosio/chain/application.hpp>
 #include <eosio/chain/exceptions.hpp>
 
+#include <thread>
```

Comment:
This all seems overly complicated. 
You should be able to simply add a mutex for the timer and call cancel on it in `stop()`. Or move the thread and ctx into file_space_handler and order it so it destructs appopriately.
Also please update to use `named_thread_pool`  of size 1.